src/algorithms/digi/TOFPulseDigitization.cc

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Souvik Paul, Chun Yuen Tsang, Prithwish Tribedy
+// Special Acknowledgement: Kolja Kauder
+//
+// Convert ADC pulses from TOFPulseGeneration into ADC and TDC values
+
+#include <math.h>
+#include <podio/RelationRange.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <gsl/pointers>
+#include <limits>
+#include <vector>
+
+#include "TOFPulseDigitization.h"
+#include "algorithms/digi/TOFHitDigiConfig.h"
+
+
+namespace eicrecon {
+
+void TOFPulseDigitization::process(const TOFPulseDigitization::Input& input,
+                                   const TOFPulseDigitization::Output& output) const {
+  const auto [simhits] = input;
+  auto [rawhits] = output;
+
+  double thres = m_cfg.t_thres;
+  // double Vm=-0.05;
+  //  SP noted that max dE experienced by LGAD should be 0.8 keV
+  double Vm = m_cfg.Vm;
+  int adc_range = m_cfg.adc_range;
+
+  // normalized time threshold
+  // convert threshold EDep to voltage
+  double norm_threshold = -thres * adc_range / Vm;
+
+  for(const auto& pulse : *simhits) {
+    // Added by SP
+    //-------------------------------------------------------------
```

Comment:
Another point of hygiene: I would suggest removing these lines (and the terminating `//--------------------`). Souvik is already accredited in the copyright. 
```suggestion
```

Reply:
Comments removed